Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -1568,6 +1585,215 @@ def getResultSummary(self):
+                extension='txt',
+                content_type='text/plain',
+            ), UploadFileToS3(
+                'build-log.txt',
+                links={self.name: 'Full build log'},
+                content_type='text/plain',
+            )
+        ]
+
+        if rc == SUCCESS:
+            steps_to_add += [ParseStaticAnalyzerResults(), CompareStaticAnalyzerResults(), ArchiveStaticAnalyzerResults(), UploadStaticAnalyzerResults(), ExtractStaticAnalyzerTestResults()]
+        self.build.addStepsAfterCurrentStep(steps_to_add)
+
+        defer.returnValue(rc)
+
+    def getResultSummary(self):
+        status = ''
+        if self.analyzeFailed:
+            status += 'ANALYZE FAILED: '
+        status += self.name
+
+        if self.commandFailed:
+            status += ' failed to build, and'
+        status += f' found {self.bugs} issues'
+
+        if self.results != SUCCESS:
+            status += f' ({Results[self.results]})'
+
+        return {u'step': status}
+
+
+class ParseStaticAnalyzerResults(shell.ShellCommandNewStyle):
+    name = 'parse-static-analyzer-results'
+    description = ['parsing static analyzer results']
+    descriptionDone = ['parsed static analyzer results']
+    haltOnFailure = True
+    resultMsg = ''
+
+    @defer.inlineCallbacks
+    def run(self):
+        output_dir = f"smart-pointer-result-archive/{self.getProperty('buildnumber')}"
+        self.command = ['python3', 'Tools/Scripts/generate-dirty-files']
+        self.command += [os.path.join(self.getProperty('builddir'), 'build/scan-build-output')]
+        self.command += ['--output-dir', os.path.join(self.getProperty('builddir'), output_dir)]
+        self.command += ['--build-dir', os.path.join(self.getProperty('builddir'), 'build')]
+
+        self.log_observer = logobserver.BufferLogObserver()
+        self.addLogObserver('stdio', self.log_observer)
+
+        rc = yield super().run()
+        if rc != SUCCESS:
+            return defer.returnValue(rc)
+
+        log_text = self.log_observer.getStdout()
+        index = log_text.rfind('Total')
+        self.resultMsg = log_text[index:]
+
+        return defer.returnValue(rc)
+
+    def evaluateCommand(self, cmd):
+        if cmd.rc != 0:
+            self.commandFailed = True
+            return FAILURE
+        return SUCCESS
+
+    def getResultSummary(self):
+        status = ''
+        if self.resultMsg:
+            status += f' Issues by project: {self.resultMsg}'
+        if self.results != SUCCESS:
+            status += f'{self.name} ({Results[self.results]})'
+
+        return {u'step': status}
+
+
+class CompareStaticAnalyzerResults(shell.ShellCommand):
+    name = 'compare-static-analyzer-results'
+    description = ['comparing static analyzer results']
+    descriptionDone = ['compared static analyzer results']
+    resultMsg = ''
+
+    @defer.inlineCallbacks
+    def run(self):
```

Comment:
Note that although you've built this like a new-style class, you're still inheriting from `shell.ShellCommand`